openupgrade_scripts/scripts/mrp/14.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_column_copies = {
+    "mrp_production": [
+        ("priority", None, None),
+        ("state", None, None),
+    ],
+}
+
+
+def fill_mrp_routing_workcenter_bom_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mrp_routing_workcenter
+        ADD COLUMN bom_id integer""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mrp_routing_workcenter
+        ADD COLUMN old_routing_workcenter_id integer""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mrp_routing_workcenter
+        ALTER COLUMN routing_id DROP NOT NULL"""
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE mrp_routing_workcenter
+        ALTER COLUMN batch DROP NOT NULL"""
```

Comment:
@MiquelRForgeFlow to avoid error when running pre-commit
```suggestion
        ALTER COLUMN batch DROP NOT NULL""",
```